### PR TITLE
Libvirt RNG, remove test cancel due to bug

### PR DESCRIPTION
**Pull request by vasekhodina**
Previously there was a test cancel step that stopped the test when multiple RNG devices were created. THis was done due to an existing bug. The bug is resolved and now this commit removes the test cancel.

### CODE REVIEW

**Comment by yanqzhan** [comment]
Looks good enough to resolve the tpm-rng issue.
But from L361-L372, I remember in s390 there's also an extra rng device that's not virtio_rng. (Sebas didn't modify codes from L418 maybe because s390 doesn't test this multiple rng case.) So how about only counting 'virtio_rng' for rng_avail and switch(L421) to another one from them?  @dzhengfy How do you think?

**Review by yanqzhan — APPROVED** [review]
Thanks! LGTM now.

File: libvirt/tests/src/libvirt_rng.py [hunk 1]
```diff
@@ -417,16 +417,20 @@ def check_guest(session, expect_fail=False,
...
             if session.cmd_status(cmd, timeout=timeout):
                 test.fail("Failed to read the random device")
 
+    def compare_device_numbers(rng_dev):
+        if vm_xml.VMXML.get_devices("tpm"):
+            rng_dev.remove("tpm-rng-0")
+        if "trng" in rng_dev:
+            rng_dev.remove("trng")
+        if len(rng_dev) != device_num:
+            test.fail("Number of rng devices defined and available does not match")
```
Comment (dzhengfy) [inline libvirt/tests/src/libvirt_rng.py:432]: I would like to have more detailed information to be included in the test.fail, like rng_dev list info and device_num value.  They are helpful for debugging. Thanks

**Review by dzhengfy — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/libvirt_rng.py (hunk 1 above)
Comment (vasekhodina) [reply libvirt/tests/src/libvirt_rng.py]: I've added more debug output.

File: libvirt/tests/src/libvirt_rng.py [hunk 2]
```diff
@@ -417,16 +417,28 @@ def check_guest(session, expect_fail=False,
...
+        :param rng_dev: List of names of RNG devices
+        """
+        if vm_xml.VMXML.get_devices("tpm"):
+            rng_dev.remove("tpm-rng-0")
+        if "trng" in rng_dev:
+            rng_dev.remove("trng")
+        if len(rng_dev) != device_num:
+            test.fail(("Number of rng devices defined and available does not match.\n"
+                       "Rng devices: %s\n"
+                       "Number of devices: %i"), rng_dev, device_num)
```
Comment (dzhengfy) [inline libvirt/tests/src/libvirt_rng.py:440]: The statement is incorrect. It should be 
```
test.fail("Number of rng devices defined and available does not match.\n"
                       "Rng devices: %s\n"
                       "Number of devices: %i" % (rng_dev, device_num))
```

Comment (vasekhodina) [reply libvirt/tests/src/libvirt_rng.py]: Fixed.

**Review by dzhengfy — APPROVED** [review]

**Comment by dzhengfy** [comment]
@vasekhodina , there is a fix for this PR. FYI. https://github.com/autotest/tp-libvirt/pull/4767
I am not sure if this pr did pass for your test because there was no result provided here.

**Pull request merged**